search-app/src/cmr/search/services/parameters/parameter_validation.clj

```diff
@@ -552,6 +552,33 @@
       [(format "Collection parameter include_facets must take value of true, false, or v2, but was [%s]"
                include-facets)])))
 
+(defn- not-all-positive-integer-values?
+  "Returns the first value in string-values that can not be converted to positive integer."
+  [string-values] 
+  (some #(when-not (and (integer? %) (< 0 %)) %) 
+        (map #(if (and (not (sequential? %))
+                       (not (clojure.string/blank? %)))   
+                (read-string %)
+                ;; not returning % here because of the case when the only non-integer value is nil. 
+                "some-string")
+             string-values))) 
+
+(defn- collection-facets-size-validation
+  "Validates that the facets-size parameter has a value positive integer value."
+  [concept-type params]
+  (when-let [facets-size (:facets-size params)]
+    (when (or (not (map? facets-size))
+              (not-all-positive-integer-values? (vals facets-size))) 
+      [(format "Collection parameter facets_size must be a map with all positive integer strings, but was [%s]"
```

Comment:
Change the message to be more readable by giving an example of a sample facets_size parameter, not "...a map with all positive integer...", as the user will not pass in a map and would probably be confused by the message.

Reply:
Good catch, I was just thinking about how to improve this message. Done.